File: Algorithms/Sorting_Algorithms/Pancake_Sort/Pancake_Sort.js
function pancakeSort(arr){
  findMaxIndex = (arr2, k) => {
    let max = -Infinity 
    let maxIndex = 0
    
    for(let i = 0; i < k; i++){
        if(arr2[i] >= max){
          max = arr2[i]
          maxIndex = i
        }       
    }
    return maxIndex
  }
  flip = (arr3, k2) => {
    let i = 0
       
    while (i < k2) {
      let temp = arr3[k2]
      arr3[k2] = arr3[i]
      arr3[i] = temp
      i++
      k2--
    }
    return arr3
  }
  let i = arr.length
  while(i > 1){
  	let maxIndex = findMaxIndex(arr, i)
  	if(maxIndex !== i - 1){
  		flip(arr, maxIndex)
  		flip(arr, i - 1)
  	}
  i--
  }
  return arr
}

pancakeSort([4,3,1,2,5,6]);//[1,2,3,4,5,6]

File: Algorithms/Sorting_Algorithms/Pancake_Sort/Pancake_Sort.cs
// C# program to implement stooge sort 
    using System;

    class Program
    {
        /// <summary>
        /// Main application function.
        /// </summary>
        public static void Main()
        {
            int[] arr = { -32, 41, 6, -2, 16, 0, 23 };

            // Calling pancake Sort function 
            // to sort the array 
            PancakeSort(arr);

            // Display the sorted array 
            for (int i = 0; i < arr.Length-1; i++)
                Console.Write(arr[i] + " ");
        }

        /// <summary>
        /// Runs pancake sort on provided array
        /// </summary>
        /// <param name="array">array to be sorted</param>
        public static int[] PancakeSort(int[] array)
        {
            for (var subArrayLength = array.Length - 1; subArrayLength >= 0; subArrayLength--)
            {
                // get the position of the maximum element of the subarray
                var indexOfMax = IndexOfMax(array, subArrayLength);
                if (indexOfMax != subArrayLength)
                {
                    // flip the array to the maximum element index
                    // the maximum element of the subarray will be located at the beginning
                    Flip(array, indexOfMax);
                    // flip the entire subarray
                    Flip(array, subArrayLength);
                }
            }

            return array;

        }
        /// <summary>
        /// method to get the index of the maximum subarray element
        /// </summary>
        /// <param name="array">array to be searched</param>
        /// <param name="n">Max index</param>
        /// <returns>Index of max element</returns>
        private static int IndexOfMax(int[] array, int n)
        {
            int result = 0;
            for (var i = 1; i <= n; ++i)
            {
                if (array[i] > array[result])
                {
                    result = i;
                }
            }

            return result;
        }

        /// <summary>
        /// method for flipping the array
        /// </summary>
        /// <param name="array">Array to be modified</param>
        /// <param name="end">Index of last item to be flipped</param>
        private static void Flip(int[] array, int end)
        {
            for (var start = 0; start < end; start++, end--)
            {
                var temp = array[start];
                array[start] = array[end];
                array[end] = temp;
            }
        }
    }
